### ISSUE TITLE
adopt systemProperty tag in javadoc

### ISSUE THREAD

**Issue by lukasj** [jakartaee/persistence#665]
to allow indexing & searching of defined properties, see https://docs.oracle.com/en/java/javase/21/docs/specs/javadoc/doc-comment-spec.html#systemproperty

**Comment by gavinking** [comment]
@lukasj two questions:

1. It's documented that this is for "system properties". I'm not quite sure what a "system property" is, but I'm not sure our configuration properties count. Do you know what the definition is?
2. Where/how were you thinking of using this tag? On which APIs?

**Comment by lukasj** [comment]
idea was to have ie `{@systemProperty jakarta.persistence.jdbc.url}` so the user can get to the definition/description of that property via javadoc search box. The main drawback here was, maybe still is that for this to work, property name must be without hyphens

**Comment by gavinking** [comment]
@lukasj The thing is that we now have string constants representing all those config properties, so I'm not sure these `@systemProperty` tags would add anything. But I'm not clear on where you thought to put them.

**Comment by lukasj** [comment]
Have it in the javadoc, for example `{@systemProperty jakarta.persistence.jdbc.url} defines JDBC URL.` Constants are nice but useless for usage in persistence.xml.

**Comment by gavinking** [comment]
_Where_ in the Javadoc?

**Comment by lukasj** [comment]
Where the definition is? Can be [here](https://github.com/jakartaee/persistence/blob/main/api/src/main/java/jakarta/persistence/Persistence.java#L313) (#L313), but I'm not sure that is the place we want to get people to, so having description at the Persistence class/property interface level, package-info or module-info is better.

**Comment by gavinking** [comment]
OK so you're thinking of it as something to go as part of  the descriptor documentation once we get up to working on that.

But my other question remains unanswered: are our properties *really* "system" properties?

**Comment by lukasj** [comment]
No. The question is - can this javadoc feature be useful for users of the persistence API?
